### PR TITLE
Adafruit Feather nrf52840 variants and UF2

### PR DESCRIPTION
**Pull request by jacobw**
Convert boards to variants and add UF2 support to Express.

### CODE REVIEW

**Comment by decsny** [comment]
wasn't there another PR about this, or is this a different set of boards, I'm not finding it

**Comment by jacobw** [comment]
> wasn't there another PR about this, or is this a different set of boards, I'm not finding it

Yes sorry forgot to link.

Replaces #76629

File: boards/adafruit/feather_nrf52840/adafruit_feather_nrf52840_flash.dtsi [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+			reg = <0x0000C000 0x00067000>;
+		};
+		slot1_partition: partition@73000 {
+			label = "image-1";
+			reg = <0x00073000 0x00067000>;
+		};
+		scratch_partition: partition@da000 {
+			label = "image-scratch";
+			reg = <0x000da000 0x0001e000>;
+		};
```
Comment (nordicjm) [inline boards/adafruit/feather_nrf52840/adafruit_feather_nrf52840_flash.dtsi:35]: no scratch partition, make slot0 2 sectors bigger than slot1 and use swap using move

Comment (jacobw) [reply boards/adafruit/feather_nrf52840/adafruit_feather_nrf52840_flash.dtsi]: Ah yeah, I'd just figured that out for another change and forgot to update this one. Done now.

File: boards/adafruit/feather_nrf52840/adafruit_feather_nrf52840_flash.dtsi [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+			label = "mcuboot";
+			reg = <0x00000000 0x0000C000>;
+		};
+		slot0_partition: partition@c000 {
+			label = "image-0";
+			reg = <0x0000C000 0x00076000>;
+		};
+		slot1_partition: partition@82000 {
+			label = "image-1";
+			reg = <0x00082000 0x00076000>;
```
Comment (nordicjm) [inline boards/adafruit/feather_nrf52840/adafruit_feather_nrf52840_flash.dtsi:30]:
```suggestion
			reg = <0x0000C000 0x00077000>;
		};
		slot1_partition: partition@82000 {
			label = "image-1";
			reg = <0x00082000 0x00075000>;
```

Comment (jacobw) [reply boards/adafruit/feather_nrf52840/adafruit_feather_nrf52840_flash.dtsi:30]: I'd copied that from the 52840dk. Does you change mean that the two partitions will be different sizes?

Comment (nordicjm) [reply boards/adafruit/feather_nrf52840/adafruit_feather_nrf52840_flash.dtsi:30]: If you use swap using move, you need a sector for the data to be moved up by, and you need space for the swap status fields, which is about a sector, so by making the changes here you get the full 0x65000 for an image, without these changes you get 0x64000

Comment (jacobw) [reply boards/adafruit/feather_nrf52840/adafruit_feather_nrf52840_flash.dtsi:30]: Got it, thanks for explaining.

Comment (jacobw) [reply boards/adafruit/feather_nrf52840/adafruit_feather_nrf52840_flash.dtsi:30]: I adjusted the start for slot1 to match.

Should the 52840dk be updated with the same changes? Partly asking as I have another PR (#77835) to move this to common.

Comment (nordicjm) [reply boards/adafruit/feather_nrf52840/adafruit_feather_nrf52840_flash.dtsi:30]: Think you've updated the wrong board

**Comment by jacobw** [comment]
@anangl ping due to no activity in a week as per PR escalation process.

I've addressed all of the feedback from @nordicjm, though didn't get approval.

**Review by nordicjm — APPROVED** [review]

**Comment by nordicjm** [comment]
Ping @anangl

**Review by anangl — APPROVED** [review]

**Pull request merged**